machine/v1beta1/types_gcpprovider.go

```diff
@@ -0,0 +1,139 @@
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GCPMachineProviderSpec is the type that will be embedded in a Machine.Spec.ProviderSpec field
+// for an GCP virtual machine. It is used by the GCP machine actuator to create a single Machine.
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type GCPMachineProviderSpec struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// UserDataSecret contains a local reference to a secret that contains the
+	// UserData to apply to the instance
+	UserDataSecret *corev1.LocalObjectReference `json:"userDataSecret,omitempty" protobuf:"bytes,2,opt,name=userDataSecret"`
+
+	// CredentialsSecret is a reference to the secret with GCP credentials.
+	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty" protobuf:"bytes,3,opt,name=credentialsSecret"`
+
+	CanIPForward       bool                   `json:"canIPForward" protobuf:"varint,4,opt,name=canIPForward"`
+	DeletionProtection bool                   `json:"deletionProtection" protobuf:"varint,5,opt,name=deletionProtection"`
+	Disks              []*GCPDisk             `json:"disks,omitempty" protobuf:"bytes,6,rep,name=disks"`
+	Labels             map[string]string      `json:"labels,omitempty" protobuf:"bytes,7,rep,name=labels"`
+	Metadata           []*GCPMetadata         `json:"gcpMetadata,omitempty" protobuf:"bytes,8,rep,name=gcpMetadata"`
+	NetworkInterfaces  []*GCPNetworkInterface `json:"networkInterfaces,omitempty" protobuf:"bytes,9,rep,name=networkInterfaces"`
+	ServiceAccounts    []GCPServiceAccount    `json:"serviceAccounts" protobuf:"bytes,10,rep,name=serviceAccounts"`
+	Tags               []string               `json:"tags,omitempty" protobuf:"bytes,11,rep,name=tags"`
+	TargetPools        []string               `json:"targetPools,omitempty" protobuf:"bytes,12,rep,name=targetPools"`
+	MachineType        string                 `json:"machineType" protobuf:"bytes,13,opt,name=machineType"`
+	Region             string                 `json:"region" protobuf:"bytes,14,opt,name=region"`
+	Zone               string                 `json:"zone" protobuf:"bytes,15,opt,name=zone"`
+	ProjectID          string                 `json:"projectID,omitempty" protobuf:"bytes,16,opt,name=projectID"`
+
+	// Preemptible indicates if created instance is preemptible
+	Preemptible bool `json:"preemptible,omitempty" protobuf:"varint,17,opt,name=preemptible"`
+}
+
+// GCPDisk describes disks for GCP.
+type GCPDisk struct {
+	AutoDelete    bool                       `json:"autoDelete" protobuf:"varint,1,opt,name=autoDelete"`
+	Boot          bool                       `json:"boot" protobuf:"varint,2,opt,name=boot"`
+	SizeGb        int64                      `json:"sizeGb" protobuf:"varint,3,opt,name=sizeGb"`
```

Comment:
You have inconsistency between other GB, you can't change json since that'd be backwards incompatible, but at least update golang type.